File: src/Ilc.Data.Models/Offer.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Ilc.Data.Models.Base;

namespace Ilc.Data.Models
{
    public class Offer : StampedEntity
    {
        public bool Active { get; set; }
        public decimal Amount { get; set; }
        public DateTimeOffset? SentAt { get; set; }
        public bool Accepted { get; set; }
        public DateTimeOffset PaymentDueAt { get; set; }
        public string Tos { get; set; }
        public bool Payed { get; set; }

        // Relations
        public virtual ICollection<Training> Trainings { get; set; }

        public virtual Customer Customer { get; set; }
    }
}

File: src/Ilc.Web/Scripts/app/model/Offer.js
Ext.define('Ilc.model.Offer', {
    extend: 'Ext.data.Model',
    
    requires: ['Ilc.model.edit.Offer'],

    fields: [
        'customer', 'active', 'createDate', 'creator', 'trainings', 'amount', 'sentAt', 'accepted',
        'paymentDueAt', 'tos', 'payed'
    ],

    getEditModel: function() {
        return Ext.create('Ilc.model.edit.Offer', {
            'id': this.get('id'),
            'active': this.get('active'),
            'payed': this.get('payed'),
            'sentAt': this.get('sentAt'),
            'accepted': this.get('accepted'),
            'amount': this.get('amount'),
            'customerId': this.get('customer').id,
            'paymentDueAt': this.get('paymentDueAt'),
            'tos': this.get('tos'),
            'trainings': R.map(R.prop('id'), this.get('trainings'))
        });
    },

    proxy: {
        type: 'rest',
        url: 'api/offers',
        extraParams: {
            format: 'json'
        },
        reader: {
            type: 'json',
            root: 'data',
            totalProperty: 'totalRecords'
        }
    }
});